pkg/controller/logstash/config.go

```diff
@@ -85,3 +116,161 @@ func defaultConfig() *settings.CanonicalConfig {
 
 	return settings.MustCanonicalConfig(settingsMap)
 }
+
+func tlsConfig(useTLS bool) *settings.CanonicalConfig {
+	if !useTLS {
+		return nil
+	}
+	return settings.MustCanonicalConfig(map[string]interface{}{
+		"api.ssl.enabled":           true,
+		"api.ssl.keystore.path":     APIKeystorePath,
+		"api.ssl.keystore.password": APIKeystoreDefaultPass,
+	})
+}
+
+// checkTLSConfig ensures logstash config `api.ssl.enabled` matches the TLS setting of API service
+// we allow disabling TLS in service and leaving `api.ssl.enabled` unset in logstash.yml, otherwise throw error
+func checkTLSConfig(config *configs.APIServer, useTLS bool) error {
+	svcUseTLS := strconv.FormatBool(useTLS)
+	sslEnabled := config.SSLEnabled
+	if (svcUseTLS == sslEnabled) || (!useTLS && sslEnabled == "") {
+		return nil
+	}
+
+	return fmt.Errorf("API Service `spec.services.tls.selfSignedCertificate.disabled` is set to `%t`, but logstash config `api.ssl.enabled` is set to `%s`", !useTLS, sslEnabled)
+}
+
+// resolveAPIServerConfig gives ExpectedAPIServer with the resolved ${VAR} value
+func resolveAPIServerConfig(cfg *settings.CanonicalConfig, params Params) (*configs.APIServer, error) {
```

Comment:
Why does this return a pointer? Initially when I read this I thought the APIServer config might be optional but it seems its not. As far as I can see it tis always populated except for the error case (where we could return an empty struct).